[Bug]: Nightly / CI failed - tests/e2e/online_serving/test_qwen3_omni_expansion.py audio-vs-text checks failing and worsening (1 case on first attempt, 4 on retry) - AssertionError: The audio content is not same as the text

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
ci env
```

</details>

### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
(not found in Buildkite step log)
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text
1ff275346923
```
</details>

### 🐛 Describe the bug

**Buildkite build:** https://buildkite.com/vllm/vllm-omni/builds/13487
**Buildkite step:** [:full_moon: Omni · Function Test with H100](https://buildkite.com/vllm/vllm-omni/builds/13487#019ff1fe-74c4-4762-882b-6c66aacfb5af) ([retry job](https://buildkite.com/vllm/vllm-omni/builds/13487#019ff3ae-b57f-4137-bc28-f35940936a0d))

**Failure kind:** pytest FAILED
**Test nodes (first attempt, 1 failed):**
- `tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_text_audio_to_text_audio_002[batch_token_64]`

**Test nodes (retry, 4 failed, finished 2026-08-12T03:29Z):**
- `tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_large_image_to_text_audio_001[default]`
- `tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_large_image_to_text_audio_001[async_chunk]`
- `tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_large_image_to_text_audio_001[batch_token_64]`
- `tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_speaker_002[default]`

**Log reason:**
AssertionError: The audio content is not same as the text

**Analysis:**
The Qwen3-Omni audio-vs-text similarity checks in this step are failing with increasing frequency, and the failing case set changes between attempts (classic flaky-but-worsening pattern):

- 2026-08-10 nightly (build 13407): original failure passed on retry; retry failed `test_large_image_to_text_audio_001[async_chunk]` instead (1 per attempt).
- 2026-08-11 nightly (build 13487, this issue): first attempt failed 1 case, the retry failed 4 cases including `test_large_image_to_text_audio_001` in **all three parametrizations**.

The symptom (transcribed audio does not match the requested text) matches the latent instruction-following failure mode analyzed in #6067 (model speaks the task description or answers the prompt instead of reading it; uncapped WER outliers on main). The same tests were previously tracked in #4518 (closed 2026-06-27). Given the trend across three consecutive nightlies, this looks like a genuine reliability regression on main rather than isolated flakes.

**Error log excerpt (representative, from the retry):**

```text
assert shorter_clean and (shorter_clean in longer_clean), (
f"The audio content is not same as the text "
f"(short-text containment check failed: "
f"text={text_output!r}, transcript={transcript!r})"
)
print(f"short-text containment check passed: {shorter_clean!r} in {longer_clean!r}")
else:
similarity = cosine_similarity_text(
transcript.lower(),
text_output.lower(),
)
print(f"similarity is: {similarity}")
>                       assert similarity > similarity_threshold, "The audio content is not same as the text"
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E                       AssertionError: The audio content is not same as the text
tests/helpers/assertions.py:669: AssertionError
```

---
*Generated from Buildkite triage of the 2026-08-12 scheduled nightly (the HTML report was not yet generated).*

--- a/tests/config/test_config_factory.py
+++ b/tests/config/test_config_factory.py
@@ -2226,6 +2226,28 @@ class TestPlatformOverrides:
         assert rocm.stages[0].enforce_eager is None
         assert rocm.stages[1].enforce_eager is True
 
+    def test_qwen3_omni_cuda_uses_thinker_rotary_custom_op(self):
+        deploy_path = Path(get_deploy_config_path("qwen3_omni_moe.yaml"))
+        pipeline = resolve_pipeline_config(
+            "qwen3_omni_moe",
+            Q3_OMNI_ALL_STAGES_HF_CONFIG,
+        )
+        assert isinstance(pipeline, PipelineConfig)
+
+        cuda = _apply_platform_overrides(load_deploy_config(deploy_path), platform="cuda")
+        cuda_stages = merge_pipeline_deploy(pipeline, cuda)
+        expected = {"custom_ops": ["+rotary_embedding"]}
+        assert cuda_stages[0].yaml_engine_args["compilation_config"] == expected
+        assert all("compilation_config" not in stage.yaml_engine_args for stage in cuda_stages[1:])
+
+        for platform in ("cpu", "musa", "npu", "rocm", "xpu"):
+            deploy = _apply_platform_overrides(
+                load_deploy_config(deploy_path),
+                platform=platform,
+            )
+            config = deploy.stages[0].compilation_config or {}
+            assert "+rotary_embedding" not in config.get("custom_ops", [])
+
     def test_minicpmo_4_5_cuda_caps_talker_kv_cache(self):
         pipeline = resolve_pipeline_config("minicpmo_4_5")
         assert isinstance(pipeline, PipelineConfig)

--- a/vllm_omni/deploy/qwen3_omni_moe.yaml
+++ b/vllm_omni/deploy/qwen3_omni_moe.yaml
@@ -69,6 +69,15 @@ stages:
       repetition_penalty: 1.1
 
 platforms:
+  # vLLM 0.27's compiled native MRoPE path can vertically fuse the thinker's
+  # Q/K RMSNorm reductions into the rotary operations and materially change
+  # Qwen3-Omni audio output (#6090). Keep MRoPE on its CUDA path instead.
+  cuda:
+    stages:
+      - stage_id: 0
+        compilation_config:
+          custom_ops: ["+rotary_embedding"]
+
   # The supported Qwen3-Omni ModelOpt checkpoint quantizes the thinker only. Clear the
   # checkpoint-level metadata for the BF16 audio stages on the verified MUSA
   # profile before vLLM performs its normal quantization auto-detection.

